tests/Makefile

```diff
@@ -45,406 +45,174 @@ ifdef ZLIB
+	test_suite_ssl		test_suite_timing		\
+	test_suite_x509parse	test_suite_x509write		\
+	test_suite_xtea		test_suite_version
+
+# Look up for associated function files
+func.test_suite_aes.ecb := test_suite_aes
+func.test_suite_aes.cbc := test_suite_aes
+func.test_suite_aes.cfb := test_suite_aes
+func.test_suite_aes.rest := test_suite_aes
+func.test_suite_arc4 := test_suite_arc4
+func.test_suite_asn1write := test_suite_asn1write
+func.test_suite_base64 := test_suite_base64
+func.test_suite_blowfish := test_suite_blowfish
+func.test_suite_camellia := test_suite_camellia
+func.test_suite_ccm := test_suite_ccm
+func.test_suite_cmac := test_suite_cmac
+func.test_suite_cipher.aes := test_suite_cipher
+func.test_suite_cipher.arc4 := test_suite_cipher
+func.test_suite_cipher.ccm := test_suite_cipher
+func.test_suite_cipher.gcm := test_suite_cipher
+func.test_suite_cipher.blowfish := test_suite_cipher
+func.test_suite_cipher.camellia := test_suite_cipher
+func.test_suite_cipher.des := test_suite_cipher
+func.test_suite_cipher.null := test_suite_cipher
+func.test_suite_cipher.padding := test_suite_cipher
+func.test_suite_ctr_drbg := test_suite_ctr_drbg
+func.test_suite_debug := test_suite_debug
+func.test_suite_des := test_suite_des
+func.test_suite_dhm := test_suite_dhm
+func.test_suite_ecdh := test_suite_ecdh
+func.test_suite_ecdsa := test_suite_ecdsa
+func.test_suite_ecjpake := test_suite_ecjpake
+func.test_suite_ecp := test_suite_ecp
+func.test_suite_error := test_suite_error
+func.test_suite_entropy := test_suite_entropy
+func.test_suite_gcm.aes128_de := test_suite_gcm
+func.test_suite_gcm.aes192_de := test_suite_gcm
+func.test_suite_gcm.aes256_de := test_suite_gcm
+func.test_suite_gcm.aes128_en := test_suite_gcm
+func.test_suite_gcm.aes192_en := test_suite_gcm
+func.test_suite_gcm.aes256_en := test_suite_gcm
+func.test_suite_gcm.camellia := test_suite_gcm
+func.test_suite_hmac_drbg.misc := test_suite_hmac_drbg
+func.test_suite_hmac_drbg.no_reseed := test_suite_hmac_drbg
+func.test_suite_hmac_drbg.nopr := test_suite_hmac_drbg
+func.test_suite_hmac_drbg.pr := test_suite_hmac_drbg
+func.test_suite_md := test_suite_md
+func.test_suite_mdx := test_suite_mdx
+func.test_suite_memory_buffer_alloc := test_suite_memory_buffer_alloc
+func.test_suite_mpi := test_suite_mpi
+func.test_suite_pem := test_suite_pem
+func.test_suite_pkcs1_v15 := test_suite_pkcs1_v15
+func.test_suite_pkcs1_v21 := test_suite_pkcs1_v21
+func.test_suite_pkcs5 := test_suite_pkcs5
+func.test_suite_pkparse := test_suite_pkparse
+func.test_suite_pkwrite := test_suite_pkwrite
+func.test_suite_pk := test_suite_pk
+func.test_suite_rsa := test_suite_rsa
+func.test_suite_shax := test_suite_shax
+func.test_suite_ssl := test_suite_ssl
+func.test_suite_timing := test_suite_timing
+func.test_suite_x509parse := test_suite_x509parse
+func.test_suite_x509write := test_suite_x509write
+func.test_suite_xtea := test_suite_xtea
+func.test_suite_version := test_suite_version
 
```

Comment:
I second the idea. However, this PR is already big enough. This can be done separately.